test/bench_cache.lua

```diff
@@ -0,0 +1,180 @@
+    },
+    name = 'primary'
+}
+
+local bucket_id_idx = {
+    type = 'TREE',
+    unique = false,
+    parts = {{path = 'bucket_id', type = 'unsigned', is_nullable = false}},
+    name = 'bucket_id'
+}
+
+local function space_init()
+    db.drop_all()
+
+    local space = {
+        engine = 'memtx',
+        is_local = true,
+        temporary = false,
+        format = table.deepcopy(helper.test_space_format())
+    }
+    table.insert(space.format, 1, {
+        name = 'bucket_id', type = 'unsigned', is_nullable = false
+    })
+
+    space.indexes = {
+        table.deepcopy(primary_index),
+        table.deepcopy(bucket_id_idx)
+    }
+    space.sharding_key = {'unsigned_nonnull', 'integer_nonnull'}
+    local schema = {
+        spaces = {
+            space = space,
+        }
+    }
+
+    return schema
+end
+
+local function run_body()
+    db.init()
+    local schema = space_init()
+    schema.spaces.space.sharding_func = {
+        body = sharding_func_body
+    }
+
+    local _, err = ddl.set_schema(schema)
+    if err then
+        print(err)
+        os.exit()
+    end
+
+    for i=1,ITERS do
+        local _, err = ddl.bucket_id('space', i)
+        if err then
+            print(err)
+            os.exit()
+        end
+    end
+
+    print("Done")
+    os.exit()
+end
+
+local function run_baseline()
+    db.init()
+    local schema = space_init()
+    schema.spaces.space.sharding_func = {
+        body = sharding_func_body
+    }
+
+    local _, err = ddl.set_schema(schema)
+    if err then
+        print(err)
+        os.exit()
+    end
+
+    for i=1,ITERS do
+        mpcrc32(i)
+        if err then
+            print(err)
+            os.exit()
+        end
+    end
+
+    print("Done")
+    os.exit()
+end
+
+local function run_name()
+    db.init()
+    local schema = space_init()
+    local sharding_func_name = 'mpcrc32'
+    rawset(_G, sharding_func_name, mpcrc32)
+    schema.spaces.space.sharding_func = sharding_func_name
+
+    local _, err = ddl.set_schema(schema)
+    if err then
+        print(err)
+        os.exit()
+    end
+
+    for i=1,ITERS do
+        local _, err = ddl.bucket_id('space', i)
+        if err then
+            print(err)
+            os.exit()
+        end
+    end
+
+    print("Done")
+    os.exit()
+end
+
+local function main()
+    local getopt = require('posix.unistd').getopt
+
+    if #arg == 0 then
+        print("Usage:", arg[0], "\n  -l - baseline benchmark",
+            "\n  -n - function name caching benchmark",
+            "\n  -b - function body caching benchmark")
+        os.exit()
+    end
+
+    for opt, _, optind in getopt(arg, 'lnb') do
+        if opt == '?' then
+            return print('unrecognized option', arg[optind -1])
+        end
+        if opt == 'l' then
+            run_baseline()
+        elseif opt == 'n' then
+            run_name()
+        elseif opt == 'b' then
+            run_body()
+        end
+    end
+end
+
+main()
```

Comment:
My personal experience suggests that when a test (including a benchmark) is not run in CI somehow, it'll become broken after some time. Feel free to ignore. You have been warned.

Reply:
moved to CI